src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -531,5 +542,707 @@ public void Add(NamespaceOrTypeSymbol symbol)
+                                    }
+                                    else
+                                    {
+                                        declarationBinder.ReportDiagnosticsIfObsolete(diagnostics, importedType, usingDirective.Name, hasBaseReceiver: false);
+
+                                        uniqueUsings.Add(importedType);
+                                        usings.Add(new NamespaceOrTypeAndUsingDirective(importedType, usingDirective, directiveDiagnostics.DependenciesBag.ToImmutableArray()));
+                                    }
+                                }
+                            }
+                            else if (imported.Kind != SymbolKind.ErrorType)
+                            {
+                                // Do not report additional error if the symbol itself is erroneous.
+
+                                // error: '<symbol>' is a '<symbol kind>' but is used as 'type or namespace'
+                                diagnostics.Add(ErrorCode.ERR_BadSKknown, usingDirective.Name.Location,
+                                    usingDirective.Name,
+                                    imported.GetKindText(),
+                                    MessageID.IDS_SK_TYPE_OR_NAMESPACE.Localize());
+                            }
+
+                            diagnostics.AddRange(directiveDiagnostics.DiagnosticBag);
+                            directiveDiagnostics.Free();
+                        }
+                    }
+
+                    uniqueUsings.Free();
+
+                    if (diagnostics.IsEmptyWithoutResolution)
+                    {
+                        diagnostics = null;
+                    }
+
+                    return new UsingsAndDiagnostics() { UsingAliases = usingAliases.ToImmutableAndFree(), UsingAliasesMap = usingAliasesMap?.ToImmutable(), UsingNamespacesOrTypes = usings.ToImmutableAndFree(), Diagnostics = diagnostics };
+                }
+            }
+
+            internal Imports GetImports(SourceNamespaceSymbol declaringSymbol, CSharpSyntaxNode declarationSyntax, ConsList<TypeSymbol>? basesBeingResolved)
+            {
+                if (_lazyImports is null)
+                {
+                    Interlocked.CompareExchange(ref _lazyImports,
+                                                Imports.Create(GetUsingAliasesMap(declaringSymbol, declarationSyntax, basesBeingResolved),
+                                                               GetUsingNamespacesOrTypes(declaringSymbol, declarationSyntax, basesBeingResolved),
+                                                               GetExternAliases(declaringSymbol, declarationSyntax)),
+                                                null);
+                }
+
+                return _lazyImports;
+            }
+
+            internal void Complete(SourceNamespaceSymbol declaringSymbol, SyntaxReference declarationSyntax, CancellationToken cancellationToken)
```

Comment:
>Complete [](start = 26, length = 8)

This is Imports.Complete with minor changes: addition/propagation of two first parameters and the code preceding ```while (true)```.